pkg/providers/aws/provider_redis.go

```diff
@@ -166,6 +179,84 @@ func (p *AWSRedisProvider) createElasticacheCluster(ctx context.Context, r *v1al
 	}}, croType.StatusMessage(fmt.Sprintf("creation successful, aws elasticache status is %s", *foundCache.Status)), nil
 }
 
+// Add Tags to AWS Elasticache
+func (p *AWSRedisProvider) TagElasticache(ctx context.Context, r *v1alpha1.Redis, stratCfg StrategyConfig, cacheClusterId string, preferredAvailabilityZone string) (types.StatusMessage, error) {
+	cacheSvcTag := elasticache.New(session.Must(session.NewSession(&aws.Config{Region: aws.String(stratCfg.Region)})))
+	cacheRegion := preferredAvailabilityZone
+	cacheRegion = cacheRegion[:(len(preferredAvailabilityZone))-1]
+	// get the account number
+	svc := sts.New(session.New())
+	inputSts := &sts.GetCallerIdentityInput{}
+	id, err := svc.GetCallerIdentity(inputSts)
+	if err != nil {
+		logrus.Error("Error GetCallerIdentity returned an error : s%", err)
+		id.Account = aws.String("")
```

Comment:
if `id.Account` is `""` can we still perform the tagging? if not, we can probably return an error here

Reply:
I think this was added to satisfy a test, returning a error will break the same test,  I will have to look into it.

Reply:
nope its fine

Reply:
Nope breaks tests in CI :( but is fine locally, looks like GetCallerIdentity is getting my local credentials which don't exist in ci.

Reply:
Do you have a link to the test that's failing? I'm not familiar with GetCallerIdentity but I'd like to understand if there are going to be any issues around this when running this in a pod instead of locally.

My cluster setup is still happening, so I can test this out more during verification once it's ready.

Reply:
https://travis-ci.com/integr8ly/cloud-resource-operator/builds/142886183
```
--- FAIL: TestAWSRedisProvider_TagElasticache (0.00s)
    --- FAIL: TestAWSRedisProvider_TagElasticache/test_tags_reconcile_completes_successfully (0.00s)
        provider_redis_test.go:427: TagElasticache() error = NoCredentialProviders: no valid providers in chain. Deprecated.
            	For verbose messaging see aws.Config.CredentialsChainVerboseErrors, wantErr false
```